### PR TITLE
Use named enum for more fields of ast::Expression

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

File: src/bin/languageserver/mod.rs [hunk 1]
```diff
@@ -516,30 +549,30 @@ impl SolangServer {
...
-                SolangServer::construct_expr(expr2, lookup_tbl, symtab, ns);
+                SolangServer::construct_expr(left, lookup_tbl, symtab, ns);
+                SolangServer::construct_expr(right, lookup_tbl, symtab, ns);
             }
 
             // Bitwise expresion
-            ast::Expression::BitwiseOr(_locs, _typ, expr1, expr2) => {
-                SolangServer::construct_expr(expr1, lookup_tbl, symtab, ns);
-                SolangServer::construct_expr(expr2, lookup_tbl, symtab, ns);
+            ast::Expression::BitwiseOr { left, right, .. } => {
```
Comment (LucasSte) [inline src/bin/languageserver/mod.rs:557]: This branch and the followings can be merged in a single branch.

File: src/bin/languageserver/mod.rs [hunk 2]
```diff
@@ -616,17 +649,17 @@ impl SolangServer {
             }
 
             //Increment-Decrement expression
-            ast::Expression::PreIncrement(_locs, _typ, _, expr1) => {
-                SolangServer::construct_expr(expr1, lookup_tbl, symtab, ns);
+            ast::Expression::PreIncrement { expr, .. } => {
```
Comment (LucasSte) [inline src/bin/languageserver/mod.rs:652]: This branch and the following can be merged.

File: src/sema/ast.rs [hunk 3]
```diff
@@ -711,17 +711,77 @@ pub enum Expression {
...
-    Power(pt::Loc, Type, bool, Box<Expression>, Box<Expression>),
-    BitwiseOr(pt::Loc, Type, Box<Expression>, Box<Expression>),
-    BitwiseAnd(pt::Loc, Type, Box<Expression>, Box<Expression>),
-    BitwiseXor(pt::Loc, Type, Box<Expression>, Box<Expression>),
-    ShiftLeft(pt::Loc, Type, Box<Expression>, Box<Expression>),
-    ShiftRight(pt::Loc, Type, Box<Expression>, Box<Expression>, bool),
+    Add {
+        loc: pt::Loc,
+        ty: Type,
+        unchecked: bool,
```
Comment (LucasSte) [inline src/sema/ast.rs:717]: Unchecked is not a self-explanatory name.

File: src/sema/ast.rs [hunk 4]
```diff
@@ -776,13 +856,13 @@ pub enum Expression {
...
-        Box<Expression>,
-        Box<Expression>,
-        Box<Expression>,
-    ),
+    ConditionalOperator {
+        loc: pt::Loc,
+        ty: Type,
+        cond: Box<Expression>,
+        left: Box<Expression>,
+        right: Box<Expression>,
```
Comment (LucasSte) [inline src/sema/ast.rs:864]:
```suggestion
        true_option: Box<Expression>,
        false_option: Box<Expression>,
```
Is this better?

**Review by LucasSte — CHANGES_REQUESTED** [review]

**Review by xermicus — APPROVED** [review]

**Review by LucasSte — APPROVED** [review]

**Pull request merged**